File: scripts/auth_token.py
#!/usr/bin/env python3
import time
import hashlib
import json
from base64 import b64encode

server_application_code = "VCM-BLUMONPAY-STG-MXN-SERVER"
server_app_key = "zsZjDKh2XvjSfQ72HyDdMIbHhw4cI2"
unix_timestamp = str(int(time.time()))
print(unix_timestamp)
uniq_token_string = server_app_key + unix_timestamp
uniq_token_hash = hashlib.sha256(uniq_token_string.encode("utf8")).hexdigest()
string = "%s;%s;%s" % (server_application_code, unix_timestamp, uniq_token_hash)
auth_token = b64encode(string.encode("utf8")).decode("utf8")
print("Content-Type: application/json")
print("")
print(json.dumps({"auth_token": auth_token}))

File: scripts/auth_token.js
#!/usr/bin/env node
// ...existing code...
const crypto = require("crypto");

const server_application_code = "VCM-BLUMONPAY-STG-MXN-SERVER";
const server_app_key = "zsZjDKh2XvjSfQ72HyDdMIbHhw4cI2";
const unix_timestamp = Math.floor(Date.now() / 1000).toString();
const uniq_token_string = server_app_key + unix_timestamp;
const uniq_token_hash = crypto
  .createHash("sha256")
  .update(uniq_token_string, "utf8")
  .digest("hex");
const tokenString = `${server_application_code};${unix_timestamp};${uniq_token_hash}`;
const token = Buffer.from(tokenString, "utf8").toString("base64");

console.log("Content-Type: application/json\n");
console.log(JSON.stringify({ token }));
// ...existing code...
